opendevin/core/exceptions.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, name=None):
         super().__init__(message)
 
 
-class LLMOutputError(Exception):
+class AgentLLMOutputError(Exception):
```

Comment:
Why did we rename this? It seems like the previous name is fine and more inclusive (the LLM doesn't need to be an Agent's LLM).

Reply:
Thanks! This is interesting... To answer the question first: this was because we have a sort of unwritten name "rule" for exceptions, to use the Agent* prefix for those exceptions that we send back to the LLM. We do that with stuff like badly formatted response in the hope it will help it self-correct next time. I think they are/were part of the agent prompt.

(from rbren)
> if there's an issue with the LLM response, we should raise an `Agent*` error, so that the message gets passed back to the LLM on the next turn instead of breaking the loop.
> 
> (Also we should rename `LLMOutputError` to something like `AgentLLMOutputError` so that we can be consistent here)

This particular treatment of this class of exceptions starts [here](https://github.com/OpenDevin/OpenDevin/blob/5ae647f5c260d1f29a1d96c4610fca485bda46d7/opendevin/controller/agent_controller.py#L297), and they're added to the event stream/history and to the agent prompt next step.

Maybe we can continue to use some prefix, different than "agent"? Or, maybe keep "agent", but this LLMOutputError is a too generic name itself, it's JSON related in practice so we could replace it with the existing [AgentMalformedActionError](https://github.com/OpenDevin/OpenDevin/blob/5ae647f5c260d1f29a1d96c4610fca485bda46d7/opendevin/core/exceptions.py#L75) or some name with "Json"

Side notes:
- I also started to document this kind of exceptions in the exceptions file, though I didn't do that consistently yet, just so that we can recognize them easier somehow
- I made this rename on the fly the first time I got to use LLMOutputError after rbren's comment, which happened to be here in this PR. I was looking the other day through the diff, to find things I can break out, but I didn't get to finish because I found and fixed a bug instead. 🐞 😅 Anyway I think this rename is not in scope and it should be easy to take out.


That said, there are a couple other odd things about Agent* errors here:
- this seems mostly a monologue and monologue-based agents thing. An Agent* error is mapped to an ErrorObservation [here](https://github.com/OpenDevin/OpenDevin/blob/5ae647f5c260d1f29a1d96c4610fca485bda46d7/opendevin/controller/agent_controller.py#L119) and thus part of agent's history, but CodeAct doesn't read it, so it doesn't send ErrorObservation back to the LLM at all. Instead it gets [here](https://github.com/OpenDevin/OpenDevin/blob/009f6f9ebc24b8cbce239f0929ac9c25bd3b055c/agenthub/codeact_agent/codeact_agent.py#L87)...
- when Agent* is raised from the summarize code however, it should be sent to the LLM, because the summarize prompt is different than the regular agent prompt. The list of texts it sends comes from [event_to_memory()](https://github.com/OpenDevin/OpenDevin/blob/5ae647f5c260d1f29a1d96c4610fca485bda46d7/opendevin/events/serialization/event.py#L71), which has text for ErrorObs. But this does mean that it's not an Agent thing anymore in a very real sense. 😅 
- this raises the question if we should let such exception (intended to be sent back) slip down from summarize to the controller... ~~maybe it makes more sense to catch and retry in summarize~~. Actually, no much of a 'maybe', I think we have to deal with it in summarize. It's the last in the event stream when it happens, so _not_ necessarily included in the "chunk" we currently summarize.

Any thoughts appreciated!

Reply:
Fixes:
- split this little refactoring out of this PR and rename all of those LLM* instead: #2508 
- handled the exception in summarization code

Reply:
Thanks!